uaclient/security.py

```diff
@@ -913,6 +919,53 @@ def _check_subscription_for_required_service(
     return False
 
 
+def _prompt_for_new_token(cfg: UAConfig) -> bool:
+    """Prompt for attach a new subscription token to the user.
+
+    :return: True if attach performed.
+    """
+    import argparse
+    from uaclient import cli
+
+    _inform_ubuntu_pro_existence_if_applicable()
+    print(status.MESSAGE_SECURITY_UPDATE_NOT_INSTALLED_EXPIRED)
+    choice = util.prompt_choices(
+        "Choose: [R]enew your subscription (at {}) [C]ancel".format(
+            BASE_UA_URL
+        ),
+        valid_choices=["r", "c"],
+    )
+    if choice == "r":
+        print(status.PROMPT_EXPIRED_ENTER_TOKEN)
+        token = input("> ")
+        print(status.colorize_commands([["ua", "detach"]]))
+        cli.action_detach(argparse.Namespace(assume_yes=True), cfg)
+
+        print(status.colorize_commands([["ua", "attach", token]]))
+        return bool(
+            0
+            == cli.action_attach(
+                argparse.Namespace(token=token, auto_enable=True), cfg
+            )
+        )
+
+    return False
+
+
+def _check_subscription_is_expired(cfg: UAConfig) -> bool:
+    from uaclient import cli
```

Comment:
I have a pedantic request fo add a docstring here representing the ':return' intent for True vs False.